gnucash/import-export/bi-import/dialog-bi-import.c

```diff
@@ -392,6 +371,28 @@ gnc_bi_import_fix_bis (GtkListStore * store, guint * fixed, guint * deleted,
 
         if (!row_deleted)
             valid = gtk_tree_model_iter_next (GTK_TREE_MODEL (store), &iter);
+        if (valid) gtk_tree_model_get (GTK_TREE_MODEL (store), &iter, ID, &id, -1);
+
+        // If this row was the last row of the invoice...
+        if (!valid || (valid && g_strcmp0 (id, running_id->str) != 0))
```

Comment:
In addition to an equal id, an empty id should also have you treat the new line as a line of the current invoice.

Reply:
I removed that logic, because I think it is bad practice to manage data records without the records' keys. But anyway, I have added it again (locally for now).